### ISSUE TITLE
Lost many changes after reload, even though the server appeared to be responding

### ISSUE THREAD

**Issue by octern** [HabitRPG/habitica#3014]
This morning I spent some time checking off, reconfiguring, and moving habits and dailies. The server seemed to be responding when I was doing this (no error messages, and some of the dailies I checked off gave drops). Then I reloaded the page, since that's still the only way to see changes in my party's outfits. I got a negative gold message showing that I'd lost some gold, and when I checked all the changes I'd made were gone. There was no negative XP message. Unfortunately, I wasn't paying enough attention to determine whether I lost the XP or the drops too. 

This is in Chrome on OS X. 
##

**Comment by octern** [comment]
Oh, I should add that this happened right after waking up my laptop in the morning. I also had to turn on my internet connection manually. I can confirm that I had an active  connection before and during the time I was making the changes, but the HRPG page was still "old." (though I didn't think that was an issue anymore.)

**Comment by deilann** [comment]
Your browser might have been in offline mode, if you had to turn on your
internet connection, even if your connection was active. I highly suggest
refreshing the page when fiddling with your network.

**Comment by octern** [comment]
As far as I know, offline mode is a browser-wide state, and I think I was accessing other web pages before using HRPG that morning. 

Beyond that, I think having the application _appear_ to communicate with the server ("updating" bar, drops appearing) while it's actually failing to do so is bad behavior. Is there any reason it shouldn't give an error message when this happens?

**Comment by deilann** [comment]
Because the site works offline -- theoretically, though, once an internet connection is established, it should sync what you have with the server.

**Comment by marphod** [comment]
I had this happen for what must have been the past 2 days, as I logged in this morning to find myself taking nearly full damage/losing all of my streaks for the past 2 days, despite logging in and doing nearly all of my dailies.  

I'm not in offline more.  And the Diadem I bought two days ago and the mount I finished then are still in my inventory. All the progress, however, is gone (and I ended upo dead; =( )

**Comment by WisTex** [comment]
This is the second time that HabitRPG has forgotten which tags I tagged something with.  I have not figured out the cause yet though.  And the weird thing is that the tasks still have the tag icon next to it, as if it were still tagged, yet they do not show up under any tags, and when you edit it, there are no tags checked.  Rechecking the tags and saving it again seems to remedy the problem, but it is annoying to have tagged everything and then it forgets.

**Comment by octern** [comment]
This exact thing happened to me too (several weeks ago -- I didn't have enough details at the time to consider reporting it). I tagged a bunch of tasks, which had previously never been tagged and didn't have tag icons. After reload the tasks had no tags on them, but the tag icon was still there.

**Comment by deilann** [comment]
That's something else. There's a ticket on it. I just don't have a moment
to look it up.

**Comment by WisTex** [comment]
I think I found the other bug about the tags:  
https://github.com/HabitRPG/habitrpg/issues/2370 Tags completely disappeared after I refreshed the page

Sorry, didn't see that earlier.

**Comment by deilann** [comment]
It's okay. :) Glad you found it.

**Comment by adamhollett** [comment]
I've had this issue as well. Chrome 38 on Windows. I leave HabitRPG in a pinned tab and use it on two PCs, one at home and one at work.

I use any mana I have in the mornings to cast backstab, and usually get a ton of gold and exp from it. This morning, for example, I got around 80 gold and exp. My exp bar then read 406/360 and I hadn't leveled up, so I thought "that's weird" and clicked the Sync button to try to fix it. I dropped back to 322/360 exp and lost my 80 gold, but my mana from using backstab didn't come back, so I had to give myself the gold back using 'fix character values'.

Other times I have used gold to buy items, then after resyncing or refreshing, the items and my gold are both gone.

I hope this was helpful. This seems like a very complicated issue. It's very annoying for me and I'd love to help get to the source of the problem.
